packages/captp/src/captp.js

```diff
@@ -137,14 +174,18 @@ export const makeCapTP = (
   let lastExportID = 0;
   // Since we decide the ids for questions, we use this to increment the
   // question key
-  let lastQuestionID = 0;
 
-  /** @type {Map<string, any>} */
-  const questions = new Map(); // chosen by us
+  /**
+   * @typedef {object} Settler
+   * @property {(result?: any) => void} resolve
+   * @property {(reason: any) => void} reject
+   * @property {(handler?: any) => void} resolvePresence
```

Comment:
I like Settlers! I have some other uses for them.

But I'm confused about the type of `resolvePresence` method. Clearly it is meant to be like `resolveWithPresence`. But the `handler` parameter here is never used. When would it be used?

Also, this is declared as returning void. However, the implementation returns the presence that `resolveWithPresence` returns, and the only call to it makes use of that presence!

Perhaps

```suggestion
   * @property {() => object} resolvePresence
```
?

If you do want to pass the `handler`, which would sometimes be forwarded to `resolveWithPresence`, what about the full type of `resolveWithPresence` including `options`? If you do go that far, perhaps `Settler` is synonymous with `HandledExecutor`?

But then the `presenceHandler` parameter of `resolveWithPresence` would need to be optional, which does not seem to make sense for `HandledExecutor`. `reason` also differs in optionality. But come to think of it, I don't understand why the `value` and `reason` of `HandledExecutor` are optional anyway?

Reply:
Does this mean we can finally stop spelling this `resPres` ;) ?